CONTRIBUTING.md

```diff
@@ -1,7 +1,142 @@
 # Contributing Guidelines
 The OSM project accepts contributions via GitHub pull requests. This document outlines the requirements for contributing to this project.
 
-## Sign Your Work
+## Pull request workflow
+
+The following sections describe how to contribute code by opening a pull request.
+
+### 1. Fork the [Open Service Mesh](https://github.com/openservicemesh/osm) repository
+
+1. Visit `https://github.com/openservicemesh/osm`.
+1. Click the `Fork` button.
+
+### 2. Clone your fork
+Set `username` to match your Github username:
+```
+export username=<github username>
+```
+
+Clone and set up your fork:
+```sh
+git clone git@github.com:$username/osm.git
+
+cd osm
+git remote add upstream git@github.com:openservicemesh/osm.git
+
+# Block accidental pushes to upstream's main branch
+git remote set-url --push upstream no_push
+
```

Comment:
This is good.
I suspect that permissions are already solving this.

Reply:
Yes it is, but in case that setting accidentally gets changed this will serve as an additional guard.